SiloHost/Program.cs

```diff
@@ -18,8 +19,39 @@ static async Task Main(string[] args)
                 .WriteTo.File(new JsonFormatter(), "logs/SiloHostLog-.log", rollingInterval: RollingInterval.Hour)
                 .CreateLogger();
 
+            var builder = new ConfigurationBuilder()
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
+            var configuration = builder.Build();
+
+            var siloPort = configuration.GetValue<int>("SiloPort");
+
+            if (siloPort == 0)
+            {
+                throw new Exception("SiloPort must be non-zero.");
+            }
+
+            var gatewayPort = configuration.GetValue<int>("GatewayPort");
+
+            if (gatewayPort == 0)
+            {
+                throw new Exception("GatewayPort must be non-zero.");
+            }
+
+            var connectionString = configuration.GetValue<string>("ConnectionString");
+
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new Exception("ConnectionString must be non-empty.");
+            }
+
             var host = new SiloHostBuilder()
-                .UseLocalhostClustering()

```

Comment:
Any idea how we can use localhostclustering on local and switch to production on production?